src/dune_rules/pkg_rules.ml

```diff
@@ -334,7 +334,21 @@ module Pkg = struct
        for build actions to run successfully, such as $PATH on systems where the
        shell's default $PATH variable doesn't include the location of standard
        programs or build tools (e.g. NixOS). *)
-    Env.extend Env.initial ~vars:(Env.Map.superpose base env)
+    let user_env = Env.initial in
+    let package_env = Env.Map.superpose base env in
+    let path_key = "PATH" in
+    let combined_path =
+      match (Env.get user_env path_key, Env.Map.find package_env path_key) with
+      | None, other -> other
+      | Some path, None -> Some path
+      | Some user_path, Some package_path ->
+        (* Put the packge PATH before the user PATH so executables from
+           packages are preferred over locally-installed executables of the
+           same name. *)
+        Some (String.concat ~sep:":" [ package_path; user_path ])
```

Comment:
Would be good to move this to `Env_path`.` We try to avoid handling raw paths in here.

Reply:
Ah yep, didn't know about `Env_path`. I've added a comment to env.ml to direct people to `Env_path` if they came to env.ml in search of operations relating to the PATH variable. Also added some documentation to clarify some of the operations for extending envs.

I added a function `Env_path.extend_env_concat_path` that merges two environments concatenating their PATHs.